client/my-sites/hosting/staging-site-card/use-is-reverting.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react';
+import { useQueryClient } from 'react-query';
+import { USE_STAGING_SITE_QUERY_KEY } from './use-staging-site';
+
+export function useIsReverting( transferStatus: string | null ): {
+	isReverting: boolean;
+	setIsReverting: ( value: boolean ) => void;
+} {
+	const [ isReverting, setIsReverting ] = useState( false );
+	const queryClient = useQueryClient();
+
+	useEffect( () => {
+		switch ( transferStatus ) {
+			case 'reverted':
+			case 'request_failure':
+				queryClient.invalidateQueries( [ USE_STAGING_SITE_QUERY_KEY ] );
+				break;
+			case null:
+				setIsReverting( false );
+				break;
+			case 'relocating_revert':
```

Comment:
We have those states in many places:
- `steps-repository/wait-for-atomic/index.tsx`
- `atomic/transfer/constants.ts`
- `woo-transfer/index.tsx`

Can we resuse it from any of those files?

And if possible, can we delete the usage of all the files, keeping only `atomic/constants`?

If not, I'd be happy to create a ticket for that.

Reply:
Good catch! It would make sense to reuse existing constants in this PR and create a separate ticket for refactoring other parts.

Reply:
Absolutely. Good point, I'm reused the classes here: https://github.com/Automattic/wp-calypso/pull/74601/commits/79e9867c1d4578d9f129064a9aea112261b00442